### PR TITLE
support for redshift auto discovery

### LINKED ISSUE #8877: Redshift auto-discovery
### What

In https://github.com/gravitational/teleport/pull/8481 we added auto-discovery for RDS instances and Aurora clusters. We should extend this for Redshift clusters as well.

Users should be able to include `redshift` type in their auto-discovery configuration and have Teleport proxy all matching Redshift databases similar to RDS:

```yaml
db_service:
  aws:
  - types: ["redshift"]
    regions: ["us-west-1"]
    tags:
      "*": "*"
```

### How

All the building blocks are there implemented in https://github.com/gravitational/teleport/pull/8481, just need to extend the database cloud watcher (`lib/srv/db/cloud/watchers`) to support Redshift in the same fashion as RDS.

### PR DESCRIPTION
**Pull request by greedy52**
closes #8877

## changes

- adds new `redshiftFetcher` to db watcher when `redshift` type is specified in the aws matchers config

## testing
Add `redshift` in db agent aws matcher types. no manual/static configuration for redshift

```
$ ./tsh db ls
Name           Description                      Labels                                                                      Connect 
-------------- -------------------------------- --------------------------------------------------------------------------- ------- 
steve-redshift Redshift cluster in ca-central-1 Env=dev,Name=steve-redshift,Owner=STeve,account-id=xxxx,region=ca-central-1         

$ ./tsh db connect --db-user stevehuang --db-name test steve-redshift
psql (14.1, server 8.0.2)
SSL connection (protocol: TLSv1.3, cipher: TLS_AES_128_GCM_SHA256, bits: 128, compression: off)
Type "help" for help.

test=#

### CODE REVIEW

File: lib/srv/db/cloud/watchers/redshift.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	return databases, nil
+}
+
+// String returns the fetcher's string description.
+func (f *redshiftFetcher) String() string {
+	return fmt.Sprintf("redshiftFetcher(Region=%v, Labels=%v)",
+		f.cfg.Region, f.cfg.Labels)
+}
+
+// getRedshiftClusters fetches all Reshift clusterss using the provided client,
```
Comment (jimbishopp) [inline lib/srv/db/cloud/watchers/redshift.go:114]:
```suggestion
// getRedshiftClusters fetches all Reshift clusters using the provided client,
```

File: docs/pages/database-access/reference/aws.mdx [hunk 2]
```diff
@@ -3,11 +3,14 @@ title: Database Access AWS IAM Reference
...
-Teleport automatically discovers and configures IAM for RDS and Aurora access
-given proper IAM permissions as described in the [RDS guide](../guides/rds.mdx).
+Teleport automatically discovers and configures IAM for RDS and Redshift access
+given proper IAM permissions as described in the [AWS RDS & Aurora
+guide](../guides/rds.mdx) and the [AWS Redshift
+guide](../guides/postgres-redshift.mdx).
 
-If you prefer to register RDS databases manually using [static configuration](./configuration.mdx)
-or [`tctl`](../guides/dynamic-registration.mdx) and manage IAM yourself, example
+If you prefer to register RDS or Redshift databases manually using [static
```
Comment (jimbishopp) [inline docs/pages/database-access/reference/aws.mdx:11]:
```suggestion
If you prefer to register RDS or Redshift databases manually using a [static
```

File: docs/pages/database-access/reference/aws.mdx [hunk 3]
```diff
@@ -52,3 +55,50 @@ arn:aws:rds-db:<region>:<account-id>:dbuser:<resource-id>/<db-user>
...
+The resource ARN string has the following format:
+
+```
+arn:aws:redshift:<region>:<account-id>:<resource>:<cluster-id>/<name>
+```
+
+Parameters:
+
+- `region`: AWS region where your Redshift cluster is deployed, or a wildcard.
+- `account-id`: ID of the AWS account the Redshift cluster is deployed in.
```
Comment (jimbishopp) [inline docs/pages/database-access/reference/aws.mdx:94]:
```suggestion
- `account-id`: ID of the AWS account where the Redshift cluster is deployed.
```

**Review by jimbishopp — COMMENTED** [review]
Looks good!

**Comment by greedy52** [comment]
@jimbishopp thanks for all the comments. Decided to separate docs vs code. This PR will be for go changes only. Will have a separate doc review PR. I will definitely include changes the comments you made here. Thanks!

**Review by r0mant — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Review by jimbishopp — APPROVED** [review]

**Pull request merged**